### ISSUE TITLE
Oxyear style repeats year twice in bibliography

### ISSUE THREAD

**Issue by ciandorr** [alex-ball/biblatex-oxref#4]
Hi, me again.  I feel like this is new: in the Oxyear style, the year shows up twice in each biubliography entry, once as the label year and then again after location and publisher.  

MWE:
```
\documentclass{article}
\usepackage[style=oxyear, natbib]{biblatex}
\usepackage{filecontents}

\begin{filecontents}{my.bib}
@book{book1,
    Author = {Test Author},
    Date = {2000},
    Title = {Title},
    Location = {Oxford},
    Publisher = {Oxford University Press},}
\end{filecontents}

\addbibresource{my.bib}

\begin{document}
Test \cite{book1}.
\printbibliography
\end{document}
```
Output:
[oxyearbaseline.pdf](https://github.com/alex-ball/biblatex-oxref/files/1709650/oxyearbaseline.pdf)

**Comment by alex-ball** [comment]
I cannot reproduce the problem with the MWE you have given, and I know why: because the bug you've found is once again triggered by using `year` instead of `date`. Thanks again for keeping me on my toes.

I should have caught that in the *Liber usualis* example in the manual, which is my test case for using `year`. What happened is that when I fixed `\iflabeldateisanydate` to also recognise `year`, it messed up my highly optimised code for the `mergedate` options, which relied on a `year`-as-`labeldatesource` to fall out the bottom of the date tests like `pubstate`.

I have rearranged the tests so they now apply `\iflabeldateisdate` first, then use `\iflabeldateisanydate` to catch `origdate` and friends.
